src/main/java/hlf/java/rest/client/service/impl/ChannelServiceImpl.java

```diff
@@ -0,0 +1,488 @@
+import hlf.java.rest.client.model.ChannelOperationType;
+import hlf.java.rest.client.model.ClientResponseModel;
+import hlf.java.rest.client.service.ChannelService;
+import hlf.java.rest.client.util.ChannelOperationStatus;
+import java.io.IOException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.*;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.hyperledger.fabric.protos.common.*;
+import org.hyperledger.fabric.sdk.*;
+import org.hyperledger.fabric.sdk.exception.CryptoException;
+import org.hyperledger.fabric.sdk.exception.InvalidArgumentException;
+import org.hyperledger.fabric.sdk.exception.ProposalException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+@Service
+@Slf4j
+public class ChannelServiceImpl implements ChannelService {
+
+  @Autowired private HFClient hfClient;
+
+  @Autowired private User user;
+
+  //    @Autowired
+  //    private KeyStoreUtil keyStoreUtil;
+
+  private final String DEAFULT_MOD_POLICY = "Admins";
+  private final String EMPTY_MOD_POLICY = "";
+  private final String GROUP_TAG_APPLICATION = "Application";
+  private final String VALUE_TAG_CONSORTIUM = "Consortium";
+  private final int INIT_VERSION = 1;
+  private final int EMPTY_VERSION = 0;
+  private final int HEADER_TYPE = 2;
+  private final String KEY_PERM_BYTE = "pemBytes";
+
+  private JsonFormat.Printer jsonPrinter = JsonFormat.printer().includingDefaultValueFields();
+
+  /** {@inheritDoc} */
+  @Override
+  public ClientResponseModel createChannel(ChannelOperationRequest channelOperationRequest) {
+    ErrorCode validationResult =
+        validateRequest(channelOperationRequest, ChannelOperationType.CREATE);
+    if (validationResult != null) {
+      throw new ChannelOperationException(validationResult);
+    }
+
+    try {
+      List<Peer> peers = getPeers(channelOperationRequest);
+      String channelName = channelOperationRequest.getChannelName();
+
+      // check if the channel exists in any peer, operation fails
+      for (Peer peer : peers) {
+        Set<String> channels = hfClient.queryChannels(peer);
+        if (channels.contains(channelName)) {
+          log.error(
+              "{} {} {}",
+              ErrorCode.FABRIC_CHANNEL_CREATION_CHANNEL_EXISTS.getReason(),
+              channelName,
+              peer);
+          throw new ChannelOperationException(ErrorCode.FABRIC_CHANNEL_CREATION_CHANNEL_EXISTS);
+        }
+      }
+
+      List<Orderer> orderers = getOrderers(channelOperationRequest);
+      Orderer orderer = orderers.get(0);
+
+      // generate new channel config
+      ChannelConfiguration channelConfiguration = newChannelConfig(channelOperationRequest);
```

Comment:
[nitpick] Looks like `ChannelOperationRequest` is passed to every method in this service class to process further. Could be a better pattern if these methods are part of the request class itself.

i.e. `channelOperationRequest.getPeers()`, `channelOperationRequest.getOrderers()` etc.